ops/testing.py

```diff
@@ -1080,6 +1094,17 @@ def get_pebble(self, socket_path: str):
         return client
 
     def planned_units(self):
+        """Simulate processing the return from goal-state into a count of planned units.
+
+        If self._planned_units is None, then we simulate what the Juju controller will do, which is
+        to report the number of peers, plus one (we include this unit in the count).
+
+        For testing purposes, a charm author can set self._planned_units explicitly by calling
+        "Harness.set_planned_units".
+        """
+        if isinstance(self._planned_units, int):
```

Comment:
This seems like an odd test vs 
```
if self._planned_units is not None:
  return self._planned_units
```

If we want to be safer for someone who accidentally calls `set_planned_units("1")` I would do that as a type check in set_planned_units rather than here.

Reply:
Good point. That was me being old fashioned, when we are, in fact, checking the type in the function signature.